adal/src/main/java/com/microsoft/aad/adal/AuthenticationResult.java

```diff
@@ -302,6 +319,33 @@ public String getIdToken() {
         return mIdToken;
     }
 
+    /**
+     * Gets if the returned token is valid in terms of extended lifetime
+     *
+     * @return True if the returned token is valid in terms of extended lifetime
+     */
+    final public boolean isExtendedLifeTimeToken() {
+        return mIsExtendedLifeTimeToken;
+    }
+
+    /**
+     * Sets the flag to indicate whether the token being returned is a token only
+     * valid in terms of extended lifetime
+     *
+     * @param isExtendedLifeTimeToken
+     */
+    final void setIsExtendedLifeTimeToken(final boolean isExtendedLifeTimeToken) {
+        mIsExtendedLifeTimeToken = isExtendedLifeTimeToken;
+    }
+
+    final void setExtendedExpiresOn(final Date extendedExpiresOn) {
+        mExtendedExpiresOn = extendedExpiresOn;
+    }
+
+    final public Date getExtendedExpiresOn() {
```

Comment:
Just to be more clear on this, we shouldn't expose getExtendedExpiresOn to developer, that's why we set expiresOn as extendedExpiresOn when returning stale token, this is to minimize the confusion to developer. All they need to know is, when they turn on the outage mode, we tell them do they get a stale token or not and what is the expiration date for the token.